Assets/Assets/Scripts/Enemy/Enemy.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Enemy : MonoBehaviour, IDamageable {
+    
+    public enum EnemyState
+    {
+        Entering,
+        Moving,
+        Attacking,
+        Dying
+    }
+
+    public int Health
+    {
+        get; set;    
+    }
+
+    public float MoveSpeed
+    {
+        get; set;
+    }
+
+    public EnemyState CurrentState
+    {
+        get; set;
+    }
+
+    public void TakeDamage(int amount)
+    {
+        this.Health -= amount;
```

Comment:
Might be better to do a `throw new NotImplementedException` for this one since it'll be inevitably overwritten by the child classes
But it's not important that it does either, so up to you